src/passes/OptimizeInstructions.cpp

```diff
@@ -974,25 +938,44 @@ struct OptimizeInstructions
   Expression* optimizeBoolean(Expression* boolean) {
     // TODO use a general getFallthroughs
     if (auto* unary = boolean->dynCast<Unary>()) {
-      if (unary && unary->op == EqZInt32) {
-        auto* unary2 = unary->value->dynCast<Unary>();
-        if (unary2 && unary2->op == EqZInt32) {
-          // double eqz
-          return unary2->value;
+      if (unary) {
+        if (unary->op == EqZInt32 || unary->op == EqZInt64) {
+          if (unary->op == EqZInt32) {
```

Comment:
why does the previous line have `EqZInt64`? I don't see where that ends up used, as the if here ignores it, and no where else later seems to use it?

Reply:
Good catch! `unary->op == EqZInt32 || unary->op == EqZInt64` is unnecessary